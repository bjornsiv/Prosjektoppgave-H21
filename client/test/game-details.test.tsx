import * as React from 'react';
import mockGameService from './mock-game-service';
import renderer from 'react-test-renderer';
import GameDetails from '../src/game-details';

jest.mock('../src/game-service', () => {
  return new mockGameService();
});

describe('Game details component tests', () => {
  test('GameDetails renders correctly with snapshot', () => {
    const tree = renderer
      .create(<GameDetails match={{ params: { id: 1 } }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});