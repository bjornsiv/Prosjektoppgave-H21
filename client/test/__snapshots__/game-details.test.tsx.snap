// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game details component tests GameDetails renders correctly with snapshot 1`] = `
<div>
  <div
    className="card rounded"
  >
    <div
      className="card-body"
    >
      <p
        className="card-title"
      >
        Game details
      </p>
      <div
        className="card-text"
      >
        <div
          className="row"
        >
          <div
            className="col-2"
          >
            <div
              className="float-start"
            >
              Game title:
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="float-start"
            >
              
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-2"
          >
            <div
              className="float-start"
            >
              Release date:
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="float-start"
            >
              2.10.1985
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-2"
          >
            <div
              className="float-start"
            >
              Genre:
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="float-start"
            >
              
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-2"
          >
            <div
              className="float-start"
            >
              Platforms:
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="float-start"
            >
              
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-2"
          >
            <div
              className="float-start"
            >
              Description:
            </div>
          </div>
          <div
            className="col"
          >
            <div
              className="float-start"
            >
              
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card rounded"
  >
    <div
      className="card-body"
    >
      <p
        className="card-title"
      >
        Average rating:
      </p>
      <div
        className="card-text"
      >
        <div
          className="col"
        >
          <div
            className="float-start"
          >
            <div />
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="float-start"
          >
            <button
              className="btn btn-dark button-darker"
              onClick={[Function]}
              type="button"
            >
              Add review
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card rounded"
  >
    <div
      className="card-body"
    >
      <p
        className="card-title"
      >
        Game reviews
      </p>
      <div
        className="card-text"
      />
    </div>
  </div>
</div>
`;
